src/utils/transaction.ts

```diff
@@ -0,0 +1,60 @@
+import { signERC2612Permit } from 'eth-permit'
+import type { Account, Chain, LocalAccount, PublicClient, WalletClient } from 'viem'
+import type { HexString } from '../types/index.js'
+import type { OperationStatus, Signer } from '../types/transaction.js'
+
+export async function* doTransaction(
+  title: string,
+  publicClient: PublicClient,
+  transactionCallback: () => Promise<HexString>
+): AsyncGenerator<OperationStatus> {
+  yield { type: 'SigningTransaction', title }
+  const hash = await transactionCallback()
+  yield { type: 'TransactionPending', title, hash }
+  const receipt = await publicClient.waitForTransactionReceipt({ hash })
+  yield { type: 'TransactionConfirmed', title, hash, receipt }
+}
+
+export async function* doSignMessage(
+  title: string,
+  transactionCallback: () => Promise<any>
+): AsyncGenerator<OperationStatus, any> {
+  yield { type: 'SigningMessage', title }
+  const message = await transactionCallback()
+  yield { type: 'SignedMessage', title, signed: message }
+  return message
+}
+
+export type Permit = {
+  deadline: number | string
+  r: string
+  s: string
+  v: number
+}
+
+export async function signPermit(
+  _: WalletClient<any, Chain, Account>,
+  signer: Signer,
+  chainId: number,
+  signingAddress: string,
+  currencyAddress: string,
+  spender: string,
+  amount: bigint
+) {
+  let domainOrCurrency: any = currencyAddress
+  if (currencyAddress.toLowerCase() === '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48') {
```

Comment:
what contract is this?

Reply:
USDC

Reply:
can I recommend storing it in a named variable instead?